### PR TITLE
feat(FX-4438): Show curated collections on search screen

### PR DESCRIPTION
**Pull request by nickskalkin**
This PR resolves [FX-4438] 

### Description

Display Artsy curated collections on search tab.

#### IOS

*Thumbnail images are broken on staging at the moment, that's why all images are grey boxes*

https://user-images.githubusercontent.com/3934579/202669279-cab81626-fffd-4537-9d12-facdd830ab29.mp4

#### Android

- [ ] I still don't have Android simulator installed, could you pair with me to install it or test changes on your device? 

### PR Checklist

- [x] I tested my changes on **iOS** 
- [ ] I tested my changes on **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Artsy curated collections list added to the Search tab

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[FX-4438]: https://artsyproduct.atlassian.net/browse/FX-4438?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Search/CuratedCollectionItem.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { IMAGE_SIZE, SearchResultImage } from "./components/SearchResultImage"
+
+interface CuratedCollectionItemProps {
+  collection: any
+}
+
+export const CuratedCollectionItem: React.FC<CuratedCollectionItemProps> = ({ collection }) => {
+  const onPress = (slug: string) => {
+    navigate(`/collection/${slug}`)
+  }
```
Comment (gkartalis) [inline src/app/Scenes/Search/CuratedCollectionItem.tsx:12]: **question:** Do we want to add analytics on this key press to measure how this feature performs?

File: src/app/Scenes/Search/CuratedCollections.tests.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import { flushPromiseQueue } from "app/tests/flushPromiseQueue"
+import { renderWithHookWrappersTL } from "app/tests/renderWithWrappers"
+import { graphql, useLazyLoadQuery } from "react-relay"
+import { act } from "react-test-renderer"
+import { createMockEnvironment } from "relay-test-utils"
+import { CuratedCollections } from "./CuratedCollections"
+
+jest.unmock("react-relay")
+
+describe("CuratedCollections", () => {
```
Comment (gkartalis) [inline src/app/Scenes/Search/CuratedCollections.tests.tsx:13]: wow 👏 👏 👏  thank you for the tests

File: src/app/Scenes/Search/Search.tsx [hunk 3]
```diff
@@ -61,6 +62,8 @@ export const Search: React.FC = () => {
   const indices = system?.algolia?.indices ?? []
   const indiceNames = indices.map((indice) => indice.name)
   const enableMaps = useFeatureFlag("AREnableMapScreen")
+  const displayCuratedCollections =
+    Platform.OS !== "ios" || useFeatureFlag("ARIosSearchTabCuratedCollections")
```
Comment (gkartalis) [inline src/app/Scenes/Search/Search.tsx:66]: **question:** that means that if its android its gonna bypass the feature flag and will be visible. Is this what we want?

If yes why do we use feature flag only for iOS? 🤔

File: src/app/Scenes/Search/CuratedCollections.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import { CuratedCollections_collections$key } from "__generated__/CuratedCollections_collections.graphql"
+import { SectionTitle } from "app/Components/SectionTitle"
+import { Join, Spacer } from "palette"
+import { graphql, useFragment } from "react-relay"
+import { CuratedCollectionItem } from "./CuratedCollectionItem"
+
+interface CuratedCollectionsProps {
+  collections: CuratedCollections_collections$key
+}
```
Comment (gkartalis) [inline src/app/Scenes/Search/CuratedCollections.tsx:9]: 🔥

**Review by gkartalis — APPROVED** [review]
Amazing work!

File: src/app/Scenes/Search/CuratedCollectionItem.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+          </Text>
+
+          <Text variant="xs" color="black60">
+            Collection
+          </Text>
+        </Flex>
+      </Flex>
+    </Touchable>
+  )
+}
```
Comment (dimatretyak) [inline src/app/Scenes/Search/CuratedCollectionItem.tsx:33]: **suggestion**: use `useFragment` here. Something like this

```javascript
...
import { CuratedCollectionItem_collection$key } from "__generated__/CuratedCollectionItem_collection.graphql"

interface CuratedCollectionItemProps {
  collection: CuratedCollectionItem_collection$key
}

export const CuratedCollectionItem: React.FC<CuratedCollectionItemProps> = ({ collection }) => {
  const data = useFragment(CuratedCollectionItemFragment, collection)

  ...
}

const CuratedCollectionItemFragment = graphql`
  fragment CuratedCollectionItem_collection on MarketingCollection {
    internalID
    slug
    title
    thumbnailImage {
      resized(width: 40) {
        url
      }
    }
  }
`
```

```javascript
// src/app/Scenes/Search/CuratedCollections.tsx
...

export const CuratedCollections: React.FC<CuratedCollectionsProps> = ({ collections }) => {
  const data = useFragment(CuratedCollectionsFragment, collections)

  return (
    <>
      <SectionTitle title="Artsy Curated Collections" />

      <Join separator={<Spacer mb={2} />}>
        {data.collections?.map((collection) => (
          <CuratedCollectionItem key={collection!.internalID} collection={collection!} />
        ))}
      </Join>
    </>
  )
}

const CuratedCollectionsFragment = graphql`
  fragment CuratedCollections_collections on Query {
    collections: curatedMarketingCollections(size: 5) {
      internalID
      ...CuratedCollectionItem_collection
    }
  }
`
```

File: src/app/Scenes/Search/CuratedCollections.tests.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  it("renders a list of curated collections", async () => {
+    const { getByText } = renderWithHookWrappersTL(<TestRenderer />, mockEnvironment)
+
+    act(() => {
+      mockEnvironment.mock.resolveMostRecentOperation({
+        data: {
+          collections,
+        },
+      })
+    })
```
Comment (dimatretyak) [inline src/app/Scenes/Search/CuratedCollections.tests.tsx:42]: **suggestion**: use `resolveMostRecentRelayOperation`. Something like this
```javascript
resolveMostRecentRelayOperation(mockEnvironment, {
  Query: () => ({
    collections,
  }),
})
```

File: src/app/Scenes/Search/CuratedCollectionItem.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+export const CuratedCollectionItem: React.FC<CuratedCollectionItemProps> = ({ collection }) => {
+  const onPress = (slug: string) => {
+    navigate(`/collection/${slug}`)
+  }
+
+  return (
+    <Touchable key={collection.internalID} onPress={() => onPress(collection.slug)}>
+      <Flex height={IMAGE_SIZE} flexDirection="row" alignItems="center">
+        <SearchResultImage imageURL={collection.thumbnailImage?.url} resultType="Collection" />
```
Comment (dimatretyak) [inline src/app/Scenes/Search/CuratedCollectionItem.tsx:17]: **question**: `thumbnailImage` can return `null`. in this case, should we use `SearchResultImage` or is it better to use some placeholder (e.g. grey box)?

File: src/app/Scenes/Search/CuratedCollectionItem.tsx (hunk 1 above)
Comment (nickskalkin) [reply src/app/Scenes/Search/CuratedCollectionItem.tsx:12]: Yes! I submitted a PR to cohesion to add new tracking event. Once it is merged - I will add tracking here

File: src/app/Scenes/Search/CuratedCollectionItem.tsx (hunk 7 above)
Comment (nickskalkin) [reply src/app/Scenes/Search/CuratedCollectionItem.tsx:17]: that's a good question. Currently if image is `null`, grey box is displayed in any case, but maybe it is better to explicitly render fallback component instead

File: src/app/Scenes/Search/Search.tsx (hunk 3 above)
Comment (nickskalkin) [reply src/app/Scenes/Search/Search.tsx:66]: Yes, feature flag is meant only for iOS, we will show curated collections in Android always

File: src/app/Scenes/Search/CuratedCollections.tests.tsx (hunk 6 above)
Comment (nickskalkin) [reply src/app/Scenes/Search/CuratedCollections.tests.tsx]: done!

File: src/app/Scenes/Search/CuratedCollectionItem.tsx (hunk 5 above)
Comment (nickskalkin) [reply src/app/Scenes/Search/CuratedCollectionItem.tsx]: done, thank you!

File: src/app/Scenes/Search/CuratedCollections.tests.tsx (hunk 6 above)
Comment (dimatretyak) [reply src/app/Scenes/Search/CuratedCollections.tests.tsx]: thanks!

File: src/app/Scenes/Search/CuratedCollectionItem.tsx (hunk 7 above)
Comment (nickskalkin) [reply src/app/Scenes/Search/CuratedCollectionItem.tsx]: actually I double-checked `SearchResultImage` component and it renders `OpaqueImageView` component underneath. It handles a case when no image is passed: https://github.com/artsy/eigen/blob/main/src/app/Components/OpaqueImageView/OpaqueImageView.tsx#L161

What do you think about keeping things like this?

**Review by dimatretyak — APPROVED** [review]
Looks great!

#### iOS
https://user-images.githubusercontent.com/3513494/202735651-44f01c0a-9b0e-4253-8bee-5531fa545e00.mp4

#### Android
https://user-images.githubusercontent.com/3513494/202738078-5df9395a-3760-4c43-a401-4ecfe870d12a.mp4

**Comment by nickskalkin** [comment]
thank you @dimatretyak for checking it on Android 👍

**Pull request merged**